src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -400,15 +403,36 @@ private LanguageExpression ConvertAccessExpression(AccessExpression expression,
             }
         }
 
-        private (LanguageExpression @base, IEnumerable<LanguageExpression> properties, bool safeAccess) ConvertModulePropertyAccess(ModuleReferenceExpression reference, PropertyAccessExpression expression) => expression.PropertyName switch
+        private (LanguageExpression @base, IEnumerable<LanguageExpression> properties, bool safeAccess) ConvertModulePropertyAccess(ModuleReferenceExpression reference, PropertyAccessExpression expression)
         {
-            // the name is dependent on the name expression which could involve locals in case of a resource collection
-            "name" => (GetModuleNameExpression(reference.Module), Enumerable.Empty<LanguageExpression>(), false),
-            "outputs" => (GetModuleReferenceExpression(reference.Module, reference.IndexContext, false),
+            switch (expression.PropertyName)
+            {
+
+                case "name":
+                    // the name is dependent on the name expression which could involve locals in case of a resource collection
+
+                    return (GetModuleNameExpression(reference.Module), Enumerable.Empty<LanguageExpression>(), false);
+
+                case "outputs":
+                    var moduleSymbol = reference.Module;
+
+                    if (context.SemanticModel.Features.SecureOutputsEnabled &&
+                        FindPossibleSecretsVisitor.FindPossibleSecretsInExpression(context.SemanticModel, moduleSymbol.DeclaringModule).Any())
```

Comment:
Not something to solve in this PR, but we should update `FindPossibleSecretsVisitor` to only emit diagnostics if the SecureOutputs feature is not enabled. I think it'll still fire even when an output is appropriately identified as secure.